dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/report/TrackerTypeReport.java

```diff
@@ -65,17 +64,12 @@ public TrackerTypeReport(
     this.trackerType = trackerType;
     this.stats = stats;
     this.sideEffectDataBundles = sideEffectDataBundles;
-
-    if (entityReport != null) {
-      for (Entity entity : entityReport) {
-        this.entityReport.put(entity.getIndex(), entity);
-      }
-    }
+    this.entityReport = entityReport;
   }
 
   @JsonProperty("objectReports")
   public List<Entity> getEntityReport() {
-    return new ArrayList<>(entityReport.values());
+    return entityReport.stream().sorted(Comparator.comparingInt(Entity::getIndex)).toList();
```

Comment:
Do you think its worth adding a unit test for this? I know we have the integration test, just wondering if that would be more obvious to spot where the error might come from.